File: Sheet 3 (Loops 1)/N.py
a = int (input())
b = 1
while True:
    a+=1
    ye_st = str(a)
    if ye_st[0] != ye_st[1] and ye_st[0] != ye_st[2] and  ye_st[0] != ye_st[3] and ye_st[1] != ye_st[2] and ye_st[1] != ye_st[3] and ye_st[2] != ye_st[3]:
        break
print(a)

File: Recursion/N.cpp
#include <bits/stdc++.h>
using namespace std;

set<string> v;
void getResult(string input, int l, int r){
    if (l == r){
        v.insert(input);
        return;
    }
    for (int i = l; i <= r; i++){
        swap(input[l], input[i]);
        getResult(input, l + 1 , r);
        swap(input[l], input[i]);
    }
    return;
}

int main(){
    string s; cin >> s;
    int n = s.length();
    getResult(s, 0, n - 1);
    cout << v.size() << '\n';
    for (auto i : v) {cout << i << '\n';}
    return 0;
}
